### PR TITLE
Added message of incorrect registration

### LINKED ISSUE #231: Message box "Incorrect registration"
Create new windows that will be showed when user already exists/incorrect repeated password/etc (based on returned codes from RegistrationUnit::RegistrationCodes).
When client click on registration button, his entered data is being validated. We need inform him what is wrong with his entered data (login, email and password). Now informing takes place using the console, but it is not the best way. We need use some widgets for it.

### PR DESCRIPTION
**Pull request by Agrifon**

### CODE REVIEW

**Comment by Orange-hanter** [comment]
1. Please make the window more readable
2. Use the coordinates of the parent window. I have incorrect behaviour in dual monitor mode.

**Comment by Agrifon** [comment]
For some reason, the test does not pass =(

File: CMakeSettings.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿{
+  "configurations": [
+    {
+      "name": "x64-Debug",
+      "generator": "Ninja",
```
Comment (f1nal3) [inline CMakeSettings.json:5]: Do we need this file?

File: Client/Client.Qt/Widgets/PopupWidget.hpp [hunk 2]
```diff
@@ -45,6 +49,9 @@ class PopupWidget : public QWidget
     void hideEvent(QHideEvent* event) override;
 
 private:
+    std::unique_ptr<QLabel> _messageText;
     std::unique_ptr<Menu> _innerMenu;
     bool                  _deleteOnHide = false;
 };
+
+void messageOut(std::shared_ptr<PopupWidget> popupWidget, QString text);
```
Comment (f1nal3) [inline Client/Client.Qt/Widgets/PopupWidget.hpp:57]: newline at the end of file

File: Client/Client.Qt/Widgets/PopupWidget.cpp [hunk 3]
```diff
@@ -59,3 +69,10 @@ void PopupWidget::setMenu(std::unique_ptr<Menu> menu)
     _innerMenu->setTriggeredCallback([=](const CallbackData&) { hide(); });
 }
 void PopupWidget::setDeleteOnHide(bool deleteOnHide) { _deleteOnHide = deleteOnHide; }
+
+void messageOut(std::shared_ptr<PopupWidget> popupWidget, QString text)
+{
+    popupWidget->resize(400, 150);
+    popupWidget->setMessage(text);
+    popupWidget->popup(QPoint(960 - 400 / 2, 540 - 150 / 2));  //nahardkodil, ny ya i negodyai
+}
```
Comment (f1nal3) [inline Client/Client.Qt/Widgets/PopupWidget.cpp:78]: newline at the end of file

File: Client/Client.Qt/Widgets/PopupSystemMessage.hpp [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    public:
+        explicit PopupSystemMessage(QWidget* parent = nullptr);
+
+    public slots:
+        void setPopupText(const QString& text);
+        void show();
+
+};
+
+#endif  // POPUP_SYSTEM_MESSAGE_H
```
Comment (f1nal3) [inline Client/Client.Qt/Widgets/PopupSystemMessage.hpp:43]: newline at the end of file

File: Client/Client.Qt/Widgets/PopupSystemMessage.cpp [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+{ 
+    popupOpacity = opacity;
+
+    setWindowOpacity(opacity);
+}
+
+float PopupSystemMessage::getPopupOpacity() const 
+{ 
+    return popupOpacity; 
+}
```
Comment (f1nal3) [inline Client/Client.Qt/Widgets/PopupSystemMessage.cpp:110]: newline at the end of file

File: Client/Client.Qt/Pages/RegistrationPage.hpp [hunk 6]
```diff
@@ -29,13 +31,14 @@ private slots:
...
+    std::unique_ptr<FlatInput>   _repeatPasswordInput;
+    std::unique_ptr<FlatButton>  _registrationButton;
+    std::unique_ptr<FlatButton>  _backButton;
+    std::unique_ptr<LogoWidget>  _logoWidget;
+    std::unique_ptr<PopupSystemMessage> _popupSystemMessage;
 
-    std::unique_ptr<QShortcut>  _registrationHotkey;
-};
+    std::unique_ptr<QShortcut>   _registrationHotkey;
+};
```
Comment (f1nal3) [inline Client/Client.Qt/Pages/RegistrationPage.hpp:44]: newline at the end of file

File: CMakeSettings.json [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+      "configurationType": "Debug",
+      "inheritEnvironments": [ "msvc_x64_x64" ],
+      "buildRoot": "${projectDir}\\out\\build\\${name}",
+      "installRoot": "${projectDir}\\out\\install\\${name}",
+      "cmakeCommandArgs": "",
+      "buildCommandArgs": "",
+      "ctestCommandArgs": ""
+    }
+  ]
+}
```
Comment (f1nal3) [inline CMakeSettings.json:15]: newline at the end of file

**Review by f1nal3 — CHANGES_REQUESTED** [review]

File: Client/Client.Qt/Widgets/PopupWidget.hpp (hunk 2 above)
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/PopupWidget.hpp:57]: Do you use this function anywhere? Looks like no. Why we should keep it?

File: README.md [hunk 8]
```diff
@@ -22,4 +22,5 @@ See @ref gettingstarted for more info on build setup and installation.
 
  See @ref localserversetup for a guide on running the server on your local machine.
 
+
```
Comment (Orange-hanter) [inline README.md:25]: Remove please

File: Client/Client.Qt/Widgets/PopupWidget.cpp [hunk 9]
```diff
@@ -31,6 +31,16 @@ void PopupWidget::hideEvent(QHideEvent*)
     if (_deleteOnHide) deleteLater();
 }
 
+void PopupWidget::setMessage(QString text) 
+{ 
+    if (!_messageText)
+    {
+        _messageText = std::make_unique<QLabel>(text, this);
```
Comment (Orange-hanter) [inline Client/Client.Qt/Widgets/PopupWidget.cpp:38]: In case of uninitialized __messageText_ you are setting the _text_ value two times.

**Review by Orange-hanter — CHANGES_REQUESTED** [review]

**Review by Orange-hanter — APPROVED** [review]

**Review by slavexx — APPROVED** [review]

File: CMakeSettings.json (hunk 1 above)
Comment (f1nal3) [reply CMakeSettings.json:5]: It is considered a good gesture to somehow comment on the resolved code review issues.

**Review by Orange-hanter — APPROVED** [review]

**Review by f1nal3 — APPROVED** [review]

**Pull request merged**